ecmd-core/capi/ecmdDataBuffer.C

```diff
@@ -2611,6 +2617,9 @@ uint32_t ecmdDataBuffer::shareBuffer(ecmdDataBuffer* i_sharingBuffer)
     rc = ecmdDataBufferBase::shareBuffer(i_sharingBuffer);
     if (rc) return rc;
 #ifndef REMOVE_SIM
+    if (i_sharingBuffer->iv_DataStr != NULL) {
+      delete[] i_sharingBuffer->iv_DataStr;
```

Comment:
Do you need to add an assign to NULL here or does the delete do that automatically?

Reply:
Only did the delete here as it is assigned a new value on line 2623 below.

Reply:
Ahh.. my bad.  When I read 2623 I flipped the assignment and thought you were assigning the deleted and undefined i_sharingBuffer to iv_DataStr.  Your way makes sense.